src/main/java/io/electrum/prepaidutility/api/EventsResource.java

```diff
@@ -0,0 +1,77 @@
+package io.electrum.prepaidutility.api;
+
+import io.electrum.prepaidutility.model.ErrorDetail;
+import io.electrum.prepaidutility.model.PurchaseResponse;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import io.swagger.annotations.Authorization;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.container.AsyncResponse;
+import javax.ws.rs.container.Suspended;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Request;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.core.UriInfo;
+
+@Path(PrepaidUtilityApi.API_BASE_PATH)
+@Api(description = "the events API", authorizations = { @Authorization("httpBasic") })
+public abstract class EventsResource {
+
+   protected abstract IEventsResource getResourceImplementation();
+
+   public static final String PATH = "/events";
+
+   public class NotifyTokenPurchase {
+      public static final String NOTIFY_TOKEN_PURCHASE = "notifyTokenPurchase";
+      public static final int SUCCESS = 202;
+      public static final String PATH = EventsResource.PATH + "/tokenPurchases/{" + PathParameters.PURCHASE_ID + "}";
+
+      public class PathParameters {
+         public static final String PURCHASE_ID = "purchaseId";
+      }
+   }
+
+   @POST
+   @Path(NotifyTokenPurchase.PATH)

```

Comment:
since the resource path is already annotated against the whole resource we just need the relative path:
```suggestion
   @Path(NotifyTokenPurchase.RELATIVE_PATH)
```